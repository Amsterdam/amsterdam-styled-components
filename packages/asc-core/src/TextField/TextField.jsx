import React from 'react'
import PropTypes from 'prop-types'
import injectSheet from 'react-jss'
import Input from '../Input/Input'
import FormLabel from '../FormLabel/FormLabel'

const styles = {
  root: {},
  error: {
    borderColor: 'red',
    color: 'red',
  },
}

const TextField = ({ classes, error }) => {
  const states = { error }
  return (
    <div className={classes.default}>
      <FormLabel {...states}>Labeltje!</FormLabel>
      <Input {...states} />
    </div>
  )
}

FormLabel.defaultProps = {
  error: false,
}

FormLabel.propTypes = {
  error: PropTypes.bool,
}

export default injectSheet(styles)(TextField)
