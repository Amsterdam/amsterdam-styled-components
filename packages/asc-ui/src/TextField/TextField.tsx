import * as React from 'react'
import { FormLabel, Input } from '@amsterdam/asc-core'

interface IProps {
  error?: boolean
}

const TextField = ({ error = false }: IProps) => {
  const states = { error }
  return (
    <div>
      <FormLabel {...states}>Labeltje!</FormLabel>
      <Input />
    </div>
  )
}

export default TextField
