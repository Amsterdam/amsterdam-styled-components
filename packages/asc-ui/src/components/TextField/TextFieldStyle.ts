import styled, { css } from '@datapunt/asc-core'
import ButtonStyle from '../Button/ButtonStyle'
import { themeSpacing } from '../../utils'

export type Props = {
  hasLabel?: boolean
}

const TextFieldStyle = styled.div<Props>`
  position: relative;
  ${({ hasLabel }) =>
    hasLabel &&
    css`
      margin-top: ${themeSpacing(2)};
    `}

  & > ${ButtonStyle} {
    position: absolute;
    top: 5px;
    right: 5px;
  }
`

export default TextFieldStyle
