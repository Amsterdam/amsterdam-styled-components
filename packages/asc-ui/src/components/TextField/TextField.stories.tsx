import * as React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'
import TextField from './TextField'

const props = {
  onBlur: action('onBlur'),
  onFocus: action('onFocus'),
  onKeyDown: action('onKeyDown'),
  value: '',
}

const TextFieldComponent: React.FC<{
  label?: string
  errorMessage?: string
}> = ({ label, errorMessage }) => {
  const [text, setText] = React.useState('')

  return (
    <TextField
      {...props}
      value={text}
      id="text-filed-id"
      label={label}
      errorMessage={errorMessage}
      onChange={e => setText(e.target.value)}
      onClear={() => setText('')}
    />
  )
}

const TextFieldSrOnlyComponent: React.FC<{}> = () => {
  const [text, setText] = React.useState('')

  return (
    <TextField
      {...props}
      srOnly
      id="text-filed-id"
      value={text}
      onChange={e => setText(e.target.value)}
      onClear={() => setText('')}
    />
  )
}

storiesOf('Atoms/TextField', module)
  .addDecorator(storyFn => (
    <div style={{ padding: '40px 10px' }}>{storyFn()}</div>
  ))
  .add('default state', () => (
    <>
      <TextFieldComponent label="Text field label" />

      <br />
      <br />
      <em>Without label:</em>
      <TextFieldComponent />
    </>
  ))
  .add('default state with label and error', () => (
    <>
      <TextFieldComponent
        label="Text field label"
        errorMessage="Error message"
      />
      <br />
      <TextFieldComponent errorMessage="Error message only" />
    </>
  ))
  .add('default state with accesible label for screen reader', () => (
    <TextFieldSrOnlyComponent />
  ))
