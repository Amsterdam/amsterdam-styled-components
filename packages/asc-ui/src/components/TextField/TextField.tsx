import { Close } from '@amsterdam/asc-assets'
import { CSSProperties, useRef } from 'react'
import { useUID } from 'react-uid'
import { useFocus } from '../../utils/hooks'
import Button from '../Button'
import FormLabelStyle from '../FormLabelStyle'
import Icon from '../Icon'
import Input, { InputProps } from '../Input'
import TextFieldStyle from './TextFieldStyle'

export interface TextFieldProps extends InputProps {
  label?: string
  keepFocus?: boolean
  srOnly?: boolean
  error?: boolean
  labelStyle?: CSSProperties
  onClear?: () => void
  autoFocus?: boolean
}

const TextField = ({
  label,
  srOnly,
  onClear,
  value,
  keepFocus,
  blurOnEscape,
  error,
  labelStyle,
  id: idProp,
  autoFocus,
  ...otherProps
}: TextFieldProps) => {
  const uid = useUID()
  const id = idProp || uid
  const inputRef = useRef<HTMLInputElement>(null)

  useFocus(inputRef, !!autoFocus)

  return (
    <TextFieldStyle>
      <FormLabelStyle
        htmlFor={id}
        srOnly={srOnly ?? false}
        label={label}
        style={labelStyle}
      >
        {label}
      </FormLabelStyle>
      <Input
        {...{ keepFocus, value, blurOnEscape, error }}
        {...otherProps}
        id={id}
        ref={inputRef}
      />
      {onClear && value && (
        <Button
          size={30}
          variant="blank"
          type="button"
          aria-label="Close"
          onClick={() => onClear()}
        >
          <Icon>
            <Close />
          </Icon>
        </Button>
      )}
    </TextFieldStyle>
  )
}

export default TextField
