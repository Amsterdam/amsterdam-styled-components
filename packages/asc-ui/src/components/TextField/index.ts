import TextFieldStyle from './TextFieldStyle'

export { default } from './TextField'

export const TextFieldStyles = {
  TextFieldStyle,
}
