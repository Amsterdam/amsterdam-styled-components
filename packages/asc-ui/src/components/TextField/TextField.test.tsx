import { shallow } from 'enzyme'
import * as React from 'react'
import TextField from './TextField'
import { ThemeProvider } from '../../index'
import 'jest-styled-components'

describe('<TextField />', () => {
  it('should render', () => {
    const component = shallow(
      <ThemeProvider>
        <TextField />
      </ThemeProvider>,
    )
    expect(component).toMatchSnapshot()
  })
})
