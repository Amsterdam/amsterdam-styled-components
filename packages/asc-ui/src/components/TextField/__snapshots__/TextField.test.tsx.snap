// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> should render 1`] = `
<ThemeProvider
  theme={
    Object {
      "breakpoints": Object {
        "desktop": [Function],
        "desktopL": [Function],
        "laptop": [Function],
        "laptopL": [Function],
        "mobileL": [Function],
        "mobileM": [Function],
        "mobileS": [Function],
        "tablet": [Function],
      },
      "colors": Object {
        "bright": Object {
          "main": "#ffffff",
        },
        "error": Object {
          "main": "#ec0000",
        },
        "primary": Object {
          "dark": "#00387a",
          "main": "#004699",
        },
        "secondary": Object {
          "dark": "#bc0000",
          "main": "#ec0000",
        },
        "support": Object {
          "focus": "#FEC813",
          "invalid": "#EC0000",
          "valid": "#00A03C",
        },
        "tint": Object {
          "level1": "#ffffff",
          "level2": "#F5F5F5",
          "level3": "#E6E6E6",
          "level4": "#B4B4B4",
          "level5": "#767676",
          "level6": "#323232",
          "level7": "#000000",
        },
      },
      "globalStyle": "",
      "typography": Object {
        "a": Object {
          "color": "#000000",
          "fontFamily": "inherit",
          "fontSize": "1rem",
          "fontWeight": 300,
          "letterSpacing": "-0.01562em",
          "lineHeight": 1,
        },
        "fontFamily": "AvenirNextLTW01-Regular, verdana, sans-serif",
        "fontSize": "16px",
        "fontWeightHeavy": 900,
        "fontWeightLight": 300,
        "fontWeightMedium": 500,
        "fontWeightRegular": 400,
        "h1": Object {
          "color": "#000000",
          "fontFamily": "inherit",
          "fontSize": "6rem",
          "fontWeight": 300,
          "letterSpacing": "-0.01562em",
          "lineHeight": 1,
        },
        "h2": Object {
          "color": "#000000",
          "fontFamily": "inherit",
          "fontSize": "6rem",
          "fontWeight": 300,
          "letterSpacing": "-0.01562em",
          "lineHeight": 1,
        },
        "h3": Object {
          "color": "#000000",
          "fontFamily": "inherit",
          "fontSize": "6rem",
          "fontWeight": 300,
          "letterSpacing": "-0.01562em",
          "lineHeight": 1,
        },
        "h4": Object {
          "color": "#000000",
          "fontFamily": "inherit",
          "fontSize": Array [
            "
      ",
            "1.125rem",
            "
    ",
          ],
          "fontWeight": 300,
          "letterSpacing": "-0.01562em",
          "lineHeight": 1,
        },
        "h5": Object {
          "color": "#000000",
          "fontFamily": "inherit",
          "fontSize": Array [
            "
      ",
            "1.125rem",
            "
    ",
          ],
          "fontWeight": 300,
          "letterSpacing": "-0.01562em",
          "lineHeight": 1,
        },
        "h6": Object {
          "color": "#000000",
          "fontFamily": "inherit",
          "fontSize": "6rem",
          "fontWeight": 300,
          "letterSpacing": "-0.01562em",
          "lineHeight": 1,
        },
        "p": Object {
          "color": "#000000",
          "fontFamily": "inherit",
          "fontSize": "1rem",
          "fontWeight": 300,
          "letterSpacing": "-0.01562em",
          "lineHeight": 1.25,
        },
      },
    }
  }
>
  <TextField />
</ThemeProvider>
`;
